src/Microsoft.AspNetCore.Mvc.Core/ApplicationModels/ApiBehaviorConventions/ApiVisibilityConvention.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.Mvc.ApplicationModels
+{
+    /// <summary>
+    /// A <see cref="IControllerModelConvention"/> that sets Api Explorer visibility.
+    /// </summary>
+    public class ApiVisibilityConvention : IControllerModelConvention
```

Comment:
Do you have a good sense for when a convention should be action/controller based? Is it based on gut feel or is there a principle behind it?

Reply:
There were two "features" that applied to the controller (for some reason I had the visibility set at the controller level rather than on the action and setting the prefix applies to controllers). Everything else applied per action and it seemed intuitive to be as specific as possible.

Reply:
We we don't have to factor the code based on how we behave. The provider could handle that. I'm asking what **should** be the right design. What do you think?

Reply:
Standardized on using `IControllerModelConvention`. I'm still looping over the actions and setting properties on it since it doesn't seem like a justified to change the behavior as part of changing how things work